addons/stock_account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2020 ForgeFlow <http://www.forgeflow.com>
+# Copyright 2020 Andrii Skrypka
+# Copyright 2021 Tecnativa - Carlos Dauden
+# Copyright 2021 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from openupgradelib import openupgrade
+from odoo import _
+from odoo.tools.float_utils import float_compare, float_is_zero, float_round
+from odoo.addons.base.models.ir_model import query_insert
+
+_logger = logging.getLogger(__name__)
+
+# Declare global variant to avoid that it is passed between methods
+precision_price = 0
+
+
+def _prepare_common_svl_vals(move, product):
+    return {
+        "create_uid": move["write_uid"],
+        "create_date": move["date"],
+        "write_uid": move["write_uid"],
+        "write_date": move["date"],
+        "stock_move_id": move["id"],
+        "company_id": move["company_id"],
+        "product_id": move["product_id"],
+        "description": move["reference"] and "%s - %s" % (move["reference"], product.name) or product.name,
+        "unit_cost": 0.0,
+        "remaining_qty": 0.0,
+        "remaining_value": 0.0,
+    }
+
+
+def _prepare_in_svl_vals(move, quantity, unit_cost, product, is_dropship):
+    vals = _prepare_common_svl_vals(move, product)
+    vals.update({
+        "value": float_round(unit_cost * quantity, precision_digits=precision_price),
+        "unit_cost": unit_cost,
+        "quantity": quantity,
+    })
+    if product.cost_method in ("average", "fifo") and not is_dropship:
+        vals["remaining_qty"] = quantity
+        vals["remaining_value"] = vals["value"]
+    return vals
+
+
+def _prepare_out_svl_vals(move, quantity, unit_cost, product):
+    # Quantity is negative for out valuation layers.
+    quantity = -quantity
+    vals = _prepare_common_svl_vals(move, product)
+    vals.update({
+        "value": float_round(unit_cost * quantity, precision_digits=precision_price),
+        "unit_cost": unit_cost,
+        "quantity": quantity,
+        "remaining_qty": 0.0,
+        "remaining_value": 0.0,
+    })
+    return vals
+
+
+def _prepare_man_svl_vals(price_history_rec, previous_price, quantity, company, product):
+    diff = price_history_rec["cost"] - previous_price
+    value = float_round(diff * quantity, precision_digits=precision_price)
+    svl_vals = {
+        "create_uid": price_history_rec["write_uid"],
```

Comment:
why not price_history_rec["create_uid"]?

Reply:
Usually the product.price.history records are not modified and create_uid is the same as write_uid, but if it were modified it would normally be to change the price, so I think the correct user is write_uid.